### PR TITLE
[azopenaiassistants] Updates for 2024-10-01-preview

### PR DESCRIPTION
**Pull request by glecaros**

### CODE REVIEW

**Review by richardpark-msft — CHANGES_REQUESTED** [review]
Looks good, let's just update the changelog to indicate that parallel tool calling is available, and a breaking change that MessageAttachmentToolAssignment has become MessageAttachmentToolDefinition.

**Review by JeffreyRichter — APPROVED** [review]

File: sdk/ai/azopenaiassistants/CHANGELOG.md [hunk 1]
```diff
@@ -23,7 +33,7 @@
 ### Bugs Fixed
 
 - ThreadRun.RequiredAction was deserialized incorrectly, making it impossible to actually resubmit a tool output. (PR#22834)
-
+a
```
Comment (weshaggard) [inline sdk/ai/azopenaiassistants/CHANGELOG.md:36]:
```suggestion

```

File: sdk/ai/azopenaiassistants/version.go [hunk 2]
```diff
@@ -7,5 +7,5 @@
 package azopenaiassistants
 
 const (
-	version = "v0.2.1"
+	version = "v0.2.2"
```
Comment (richardpark-msft) [inline sdk/ai/azopenaiassistants/version.go:10]:
```suggestion
	version = "v0.3.0"
```

We've been doing "breaking change means rev the minor version for betas". (also, we'll need to update the changelog)

Comment (glecaros) [reply sdk/ai/azopenaiassistants/version.go:10]: fixed :)

**Review by richardpark-msft — APPROVED** [review]

**Pull request merged**